model/src/main/kotlin/Identifier.kt

```diff
@@ -114,9 +114,13 @@ data class Identifier(
 
     /**
      * Create a file system path based on the properties of the [Identifier]. All properties are encoded using
-     * [encodeOrUnknown].
+     * [encodeOr] with [emptyValue] as parameter.
      */
-    fun toPath(separator: String = "/") = components.joinToString(separator) { it.encodeOrUnknown() }
+    fun toPath(separator: String = "/", emptyValue: String = "unknown"): String {
+        require(emptyValue.isNotEmpty())
```

Comment:
Shouldn't this check already go to `encodeOr()`? Because if you replace an empty string with an empty string, you could as well omit a call to `encodeOr()` altogether.